juju_verify/verifiers/result.py

```diff
@@ -175,12 +178,75 @@ def add_partial_result(self, severity: Severity, message: str) -> None:
         self.partials.append(Partial(severity, message))
 
 
-def aggregate_results(*results: Result) -> Result:
-    """Return aggregate value of multiple results."""
-    result_list = list(results)
-    final_result = result_list.pop(0)
+def checks_executor(
+        *checks: Union[Callable, Tuple[Callable, Dict[str, Any]]]) -> Result:
+    """Executor that aggregates checks and captures errors.
+
+    This executor will accept checks as a callable function or as a tuple with first
+    object callable (check) and second as a dictionary containing parameters for
+    performing the check. The executor's output aggregates all check results into one
+    result, while the order is preserved. If the check does not return any input, then
+    the default value is used in the form:
+    `Result(OK, "<check .__ name __> check successful").
+    At the same time, if the check fails on one of the following errors
+    (JujuActionFailed, CharmException, KeyError, JSONDecodeError), it will be marked
+    as failed with the result in the form:
+    `Result(FAIL, f"{check.__name__} check failed with error: {error}"`.
+
+    Examples of use:
+
+    def check_without_parameter_1():
+        return Result(Severity.OK, "check 1 passed")
+
+    def check_without_parameter_2():
+        return Result(Severity.OK, "check 2 passed")
+
+    def check_with_parameter(resources=None):
+        if resources is None:
+            return Result(Severity.FAIL, "check 3 failed")
 
-    for result in result_list:
-        final_result += result
+        return Result(Severity.OK, f"check 3 passed ({resources})")
 
-    return final_result
+    result_1 = checks_executor(check_without_parameter_1,
+                               check_without_parameter_2,
+                               (check_with_parameter, dict(resources="DHCP")))
+
+    print(resurce_1)
```

Comment:
This should be `print(result_1)`

Reply:
I will fixed this in upcoming PR ("cli/exit-at-failed-check").